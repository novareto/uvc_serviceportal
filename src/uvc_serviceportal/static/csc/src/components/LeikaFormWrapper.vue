<template>
    <div>
      <form action="" @submit.prevent="submit">
      <JsonSchema :schema="schema" v-model="model" :wrapper="customWrapper" />
      <input type="submit" class="btn btn-primary" @click.prevent="submit">
      </form>
    <custom-wrapper> </custom-wrapper>
    </div>
</template>

<script>
import JsonSchema from '@roma219/vue-jsonschema-form'
import axios from 'axios'
import CustomWrapper from './CustomWrapper'

export default {
    props: {
        schema: {
            type: Object
        },
    },
    data() {
        return {
            model: {},
        }
    },
    computed: {
      customWrapper() {
          console.log('I AM IN ')
      return {
        componentName: 'CustomWrapper',
        props: (propName, schema) => ({
          title: schema.title || propName
        })
      }
    },
    },
    methods: {
        submit() {
            var formData = new FormData()
            for ( var key in this.model ) {
                formData.append(key, this.model[key]);
            }
            axios.post('/leikas/leika1/add', formData).then((response) => {
                     console.log(response)
                        // success callback
                    }, (response) => {
                        console.log(response)
                        // error callback
                    });
        }
    },
    components: { JsonSchema, 'custom-wrapper': CustomWrapper }
}
</script>

<style lang="scss" scoped>

</style>
