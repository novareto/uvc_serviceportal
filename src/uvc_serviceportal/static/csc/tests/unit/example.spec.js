import { mount } from '@vue/test-utils'
import LeikaFormWrapper from '@/components/LeikaFormWrapper.vue'

    /*
describe('LeikaFormWrapper.vue', () => {
  it('renders props.msg when passed', () => {
    const wrapper = mount(LeikaFormWrapper, {
      propsData: { schema }
    })
    console.log('HANS MEIER')
    expect(wrapper.text()).toMatch('msg')
  })
})
*/

test('mount a view component', () => {
      const schema = {
          "type": "object",
          "properties": {
              "firstName": {
                  "type": "string"
              }
          }
      }
    const wrapper = mount(LeikaFormWrapper, {
      propsData: { schema: schema }
    })
    expect(wrapper.html()).toMatchSnapshot()

})
