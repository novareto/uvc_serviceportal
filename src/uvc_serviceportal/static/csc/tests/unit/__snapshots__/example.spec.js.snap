// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount a view component 1`] = `
<div>
  <form action="">
    <div class="json-schema-form pure-form pure-form-stacked">
      <div class="pure-control-group"><label>firstName</label> <input type="text">
        <!---->
      </div>
    </div> <input type="submit" class="btn btn-primary">
  </form>
</div>
`;
